x-pack/plugins/enterprise_search/public/applications/shared/layout/nav_link_helpers.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { stripTrailingSlash } from '../../../../common/strip_slashes';
+
+import { KibanaLogic } from '../kibana';
+import { generateReactRouterProps, ReactRouterProps } from '../react_router_helpers';
+
+interface Params {
```

Comment:
I've been using more generic type names (e.g. `Props`, `Params`) **if** they're not getting exported or reused externally. If they're used in other files, that's when I namespace them (e.g. `PageTemplateProps`, `ReactRouterProps`).

But otherwise it's to me it's kinda like not having to namespace `isDocumentsLoading` vs `isAnalyticsLoading` - the `isLoading` var name is scoped to the file or view, so it doesn't need to be overly specific.

The day I ever start using generic single letter variable names is the day you can hit me over the head with a shovel tho, gotta draw that line somewhere :trollface: